modules/repofiles/content.go

```diff
@@ -146,13 +146,19 @@ func GetContents(repo *models.Repository, treePath, ref string, forList bool) (*
 	}
 	selfURLString := selfURL.String()
 
+	lastCommit, err := commit.GetCommitByPath(treePath)
```

Comment:
This could be quite slow depending on the version of git present and the size of the repository. 

Internally this code may need to be tied in to the last commit cache.